### PR TITLE
fix: return better error messages in the case of a bad issuer

### LINKED ISSUE #1701: zcf.saveIssuer() with bad issuer - improve error message
**What is the Problem Being Solved?**

If we call `zcf.saveIssuer()` with a bad issuer, we get the following unhelpful error message:

```
    message: 'o["getBrand"] is not a function',
```

**Description of the Design**

Improve this error message by either doing an assertion before this call (an extra trip) or catching any errors related to the issuer not having the expected API and inserting a custom, improved error message.

**Security Considerations**

If we catch errors, we should make sure that we are not suppressing errors unrelated to the issuer not having the expected API.

**Test Plan**
The ZCF tests that found the issue should confirm the error message has improved. We may want to have additional tests to ensure we aren't suppressing errors unrelated.

### LINKED ISSUE #1702: zcf.saveIssuer - improve error message if arguments reversed
**What is the Problem Being Solved?**

Currently the error message if the arguments are reversed is not as helpful as it could be:

```js
test(`zcf.saveIssuer - args reversed`, async t => {
  const { moolaKit } = setup();
  const { zcf } = await setupZCFTest();
  t.throws(() => zcf.saveIssuer('A', moolaKit.issuer), {
    message: `(an object) must be a string\nSee console for error data.`,
  });
});
```

**Description of the Design**

Mention in the error message that the arguments may have been reversed.

**Security Considerations**

None.

**Test Plan**
ZCF tests

### PR DESCRIPTION
**Pull request by katelynsills**
Closes #1701
Closes #1702

### CODE REVIEW

File: packages/zoe/src/issuerTable.js [hunk 1]
```diff
@@ -39,19 +39,26 @@ const makeIssuerTable = () => {
...
-      ]);
-      // AWAIT /////
+
+      let issuer;
+      let brand;
+      let brandIssuerMatch;
+      let amountMathKind;
+      try {
+        /** @type {[Issuer,Brand,boolean,AmountMathKind]} */
+        ([issuer, brand, brandIssuerMatch, amountMathKind] = await Promise.all([
```
Comment (Chris-Hibbert) [inline packages/zoe/src/issuerTable.js]: This type declaration isn't actually having the effect it should. If you hover over any of the variables in the next line, they show as `any`.  I'd try putting the type declarations on the variable declarations (lines 43-46)

File: packages/zoe/src/issuerTable.js [hunk 2]
```diff
@@ -60,6 +67,9 @@ const makeIssuerTable = () => {
         brandIssuerMatch,
         `issuer was using a brand which was not its own`,
       );
+      assert(issuer);
+      assert(brand);
+      assert(amountMathKind);
```
Comment (Chris-Hibbert) [inline packages/zoe/src/issuerTable.js]: should these have error messages?

**Review by Chris-Hibbert — APPROVED** [review]
This seems fine.  I added some comments that may point a direction to fixing the lint problem.  It may  be worth adding error messages on the bare asserts. Otherwise, approved.

File: packages/zoe/src/issuerTable.js (hunk 1 above)
Comment (erights) [inline packages/zoe/src/issuerTable.js]: `await` inside a `try` violates our rules. It is silly in this case anyway. Please rephrase your catch clause as a rejection handler on the promise returned by `Promise.all`.

File: packages/zoe/src/issuerTable.js [hunk 3]
```diff
@@ -39,19 +39,26 @@ const makeIssuerTable = () => {
...
+        ([issuer, brand, brandIssuerMatch, amountMathKind] = await Promise.all([
+          issuerP,
+          brandP,
+          brandIssuerMatchP,
+          amountMathKindP,
+        ]));
+        // AWAIT /////
+      } catch (err) {
+        console.error(err);
+        assert.fail(
```
Comment (erights) [inline packages/zoe/src/issuerTable.js]: Please add a TODO here to rethrow the error after adding the details as a note (for upcoming assert library). The problem, which you can't fix well until then, is that the error you're throwing no longer has the informative stack trace of the error you caught.

**Review by erights — CHANGES_REQUESTED** [review]

**Comment by katelynsills** [comment]
@erights it is this PR in particular that will benefit from `getInterfaceOf`.

**Comment by katelynsills** [comment]
Closing this in favor of a new PR when we can get `getInterfaceOf` in a different way than `vatPowers.getInterfaceOf`. See #1816

**Pull request closed without merging**